test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test.csproj

```diff
@@ -28,8 +29,23 @@
     <PackageReference Include="Moq" Version="$(MoqVersion)" />
     <PackageReference Include="xunit" Version="$(XunitVersion)" />
     <PackageReference Include="xunit.runner.visualstudio" Version="$(XunitVersion)" />
+
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="2.0.0-*" />
+
+    <!-- We need this to compile but don't want it referenced by the project.
+      It is included dynamically in CodeGeneration tests. -->
+    <ProjectReference Include="..\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Razor.Test.MvcShim.csproj">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+      <OutputItemType>Content</OutputItemType>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <Targets>Build</Targets>
+    </ProjectReference>
   </ItemGroup>
 
+  <Target Name="CopyDepsFromShims" AfterTargets="Build">
+    <Copy SourceFiles="..\Microsoft.AspNetCore.Razor.Test.MvcShim\bin\$(Configuration)\net46\Microsoft.AspNetCore.Razor.Test.MvcShim.deps.json" DestinationFolder="$(OutputPath)" />
```

Comment:
Can lift the location of the `Test.MvcShim` into an msbuild ItemGroup. That way you could re-use that property here and in the ProjectReference above. Note that an item group is more than 1. This way, when we add more shims in the future we can simply add to the ItemGroup and everything continues to work.